client/hotrod-client/src/main/java/org/infinispan/client/hotrod/logging/Log.java

```diff
@@ -399,4 +399,8 @@ public interface Log extends BasicLogger {
    @Message(value = "OpenTelemetry API is present in the classpath, but the tracing propagation is not enabled. Client context tracing will not be propagated.", id = 4111)
    void openTelemetryPropagationDisabled();
 
+   @LogMessage(level = DEBUG)
```

Comment:
I wonder if this should be ERROR now? If this exception is encountered the user has open telemetry on the classpath and has explicitly configured trace propagation.

Reply:
Right this error shouldn't happen, since we've already checked that the OpenTelemetry is present on the classpath.
It looks like a very runtime exception...

Looking at the code that is executed by the current OpenTelemetry version, it should be safe to call `W3CTraceContextPropagator.getInstance()`. 

Anyway we don't have control over this library, that's the reason I introduced the check.
In general implementing tracing I usually write code that is usually more defensive, since in any case tracing should affect the behaviour of the application.

Anyway it is OK for me to remove this check.

Reply:
Oh I am not saying to remove it. Whether we let it propagate or log the exception is up to you. I was only proposing it be higher priority log level.

Reply:
Sure, makes sense